cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -272,6 +329,8 @@ public void drop(Id.DatasetInstance instance) throws Exception {
     LOG.info("Deleting dataset {}.{}", instance.getNamespaceId(), instance.getId());
     dropDataset(instance, spec);
     publishAudit(instance, AuditType.DELETE);
+    // revoke privileges
+    authorizerInstantiator.get().revoke(instance.toEntityId());
```

Comment:
yes, the thought was we never have any privileges in the system on non-existing entities. The other scenario is that you deleted some entities, but privileges still exist. Now if someone creates an entity (potentially completely different, but with the same name), some users automatically (unintentionally?) get access to it.